src/mpid/ch4/netmod/ofi/ofi_capability_sets.h

```diff
@@ -190,7 +190,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_get_set_number(const char *set_name)
 #define MPIDI_OFI_SOURCE_BITS_SOCKETS               (0)
 #define MPIDI_OFI_TAG_BITS_SOCKETS                  (31)
 #define MPIDI_OFI_MAJOR_VERSION_SOCKETS             1
-#define MPIDI_OFI_MINOR_VERSION_SOCKETS             5
+#define MPIDI_OFI_MINOR_VERSION_SOCKETS             4
```

Comment:
I think if you actually got a libfabric version < 1.4, it wouldn't compile. I think we shifted to a hard requirement of 1.5 to support things like `FI_CONTEXT2` a long time ago.

Reply:
That is probably true. However, the version we requested also affects the libfabric behaviors. If we request for at 1.4, it will try to use pre-1.5 semantics. The `sockets` provider doesn't really support individual `MR` bits, so using post-1.5 semantics doesn't work nicely.

Reply:
🤷‍♂️ We're not seeing any of these problems internally so maybe by the time we catch up, `main` will be working. 😄

Reply:
No. That's not how mode bits work. MPI says what we support and OFI says what we have to use. In our case, we always leave the MR mode bits off for sockets.

Reply:
`sockets` *supports* `FI_MR_BASIC`, equivalent to have the 3 bits on. However, setting the 3 bits on won't turn on `FI_MR_BASIC` because `sockets` doesn't really support individual bits. Essentially, `sockets` only support pre-1.5 settings.

Reply:
@wesbland Let me summarize what the change in question does. Without this change, all the tests with `sockets` still work and pass. However, we *will not able to* test the `FI_MR_BASIC` settings. `sockets` provider will only switch to `FI_MR_BASIC` with `FI_MR_BASIC` bit set, i.e. the pre-1.5 semantics. With these changes, we can switch tests to `FI_MR_BASIC` if we set all `MPIR_CVAR_CH4_OFI_ENABLE_MR_ALLOCATED`, `MPIR_CVAR_CH4_OFI_ENABLE_MR_VIRT_ADDRESS` and ``MPIR_CVAR_CH4_OFI_ENABLE_MR_PROV_KEY` to `1`. The ability to test the latter is important since it covers some important code paths. For example, without that test, we never discover the bug in `MPL_avltree` -- https://github.com/pmodels/mpich/issues/5243

Reply:
Both runtime selection or static builds will work by default since we use the setting that all `mr_mode` bits are off. Previously we had a bug (that not using correct required fi_version) and we had a hack in the current main that to set `FI_MR_SCALED` if all bits are off. The hack (https://github.com/pmodels/mpich/pull/5262/files#r630450173) is due to not understanding the actual issue -- that we are not setting the correct the minimum provider version -- and the hack is unnecessary if we do it correctly (which this PR corrected). With the corrected behavior, however, with the setting of version 1.5 (the change in question here), we are not able to turn on the `FI_MR_BASIC` mode for `sockets` provider because it simply doesn't use individual bits. In order to be able to test `FI_MR_BASIC` (not by default but custom tests), we need request pre-1.5 behaviors.

Reply:
> In order to be able to test `FI_MR_BASIC` (not by default but custom tests), we need request pre-1.5 behaviors.

This is what I was trying to understand. It is for custom builds. It is a bit odd that the sockets capability set won't be able to take advantage of some newer features. On the other hand it is in maintenance mode and not seeing active development. I think as long as we can reliably use the tcp provider soon, then this is less of a concern.

Reply:
Let me de-bundle the changes from this PR ...

Reply:
The change is removed. Will open a separate PR for the version change after this one merges.